### PR TITLE
Add an implementation of the Porta cipher.

### PR DESCRIPTION
**Pull request by ra-kete**
Looking at the existing implementation of the Vigenere cipher, which is quite similar to the Porta cipher, has simplified this one.
Solves #40.

### CODE REVIEW

**Comment by ra-kete** [comment]
Checks fail currently due to rustfmt incompatibilities. I fixed those in #42.

**Comment by arosspope** [comment]
Okay, you might want to merge master back in so the Travis checks pass.

File: src/porta.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
+//! The Porta Cipher is a polyalphabetic substitution cipher. It was invented
+//! by Giovanni Battista della Porta, an Italian polymath, in 1563.
+//!
+//! # Algorithm
```
Comment (arosspope) [inline src/porta.rs]: Overall this is really great documentation. To make sure it's in line with the other ciphers could you please:

1) Drop the sub-headings and make the content under the `Encryption` sub-heading as a new paragraph.
2) Within the above paragraph, specify that this cipher is a reciprocal cipher (process to encrypt is same as decrypt).
3) Follow this paragraph with the substitution table - include a short sentence explaining what it is.
3) Delete the `Decryption` sub-heading.
4) Move the content in the `Encryption Example` section, to the documentation for the `encrypt` function.

**Comment by ra-kete** [comment]
Done. I dropped the `Encryption Example` part, since IMO it was redundant with the `encrypt` example code anyway.

**Comment by arosspope** [comment]
LGTM

**Pull request merged**